### PR TITLE
Switch to Helm v3

### PR DESCRIPTION
**Pull request by alex-yust**
## Description

v3 is an overall improvement over v2 and it seems right to switch to using it while the project's chart are completely compatible with v3.

major differences:
* no Tiller -> no need to install it (make helm-init)

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/guide-helm.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ This document will show you how to use `Helm` for `NSM` installation.
 Installing NSM with helm in the `nsm-system` namespace is as easy as:
 
 ```bash
-$ helm install deployments/helm/nsm
+$ helm install nsm deployments/helm/nsm
 ```
 
 *Note: in case of `Error: no available release name found` do (according to [issue](https://github.com/helm/helm/issues/4412)):*
```
Comment (jellonek) [inline docs/guide-helm.md]: This whole section can be probably removed, right?

Comment (alex-yust) [reply docs/guide-helm.md]: You're right, thanks!
- [ ] remove tiller issue troubleshooting

**Comment by jellonek** [comment]
Please rebase as fix for tests is already merged (was part of #2114).

**Comment by haiodo** [comment]
@nickolaev @edwarnicke Any idea why we not merged this PR yet?

**Pull request merged**